### PR TITLE
Fix the build on big-endian hosts and a related cleanup

### PR DESCRIPTION
**Pull request by smaeul**
This allows me to compile and run or1ksim on a powerpc64 box.

### CODE REVIEW

File: cpu/common/elf.h [hunk 1]
```diff
@@ -6,8 +6,8 @@
 #endif
 
 #ifdef WORDS_BIGENDIAN
-#define ELF_SHORT_H
-#define ELF_LONG_H
+#define ELF_SHORT_H(ps) ((unsigned short)(ps))
+#define ELF_LONG_H(ps)  ((unsigned long)(ps))
```
Comment (jeremybennett) [inline cpu/common/elf.h:10]: This changes the macro prototype, but I can't seen any change to the use of the macro below. Or is this just fixing a bug that was always there?

**Review by jeremybennett — COMMENTED** [review]
As noted elsewhere I don't actively maintain Or1ksim any longer, but I'm in principle happy with this. The important thing is that you have tested it well and that it doesn't break existing behavior. As well as the Or1ksim tests, running something like GCC regression into Or1ksim is a good check of robust behavior, at least of the CPU model.

I'd appreciate comment from @juliusbaxter before committing.

File: cpu/common/elf.h (hunk 1 above)
Comment (smaeul) [reply cpu/common/elf.h:10]: It is fixing a bug that was always there. The old definition made the macro a no-op, but the cast is actually necessary, or else GCC warns for printf usage:
```
../../../cpu/common/parse.c:715:20: error: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'Elf32_Addr' {aka 'unsigned int'} [-Werror=format=]
  715 |           PRINTFQ (" vaddr: 0x%.8lx,", ELF_LONG_H (elf_spnt->sh_addr));
      |                    ^~~~~~~~~~~~~~~~~~             ~~~~~~~~~~~~~~~~~~~
      |                                                            |
      |                                                            Elf32_Addr {aka unsigned int}
```
The new prototype now matches the little-endian version of the macro.

**Comment by smaeul** [comment]
I was able to boot Linux to userspace inside or1ksim. I can try to figure out how to run gcc tests as well.

File: cpu/common/elf.h (hunk 1 above)
Comment (jeremybennett) [reply cpu/common/elf.h:10]: Good work

**Comment by jeremybennett** [comment]
I'm about to disappear on vacation and then to Embedded World. We'll give @juliusbaxter a couple of weeks to comment and then if there are no other comments, I'll merge the PR.

**Comment by stffrdhrn** [comment]
I'll also look at this.  I can just try to run both patches on my system as well .

For gcc tests it uses dejagnu. For that I have a site.exp here. https://github.com/stffrdhrn/or1k-utils

It's setup to just use sim or user mode qemu.  To use or1ksim I think we used telnet before. I haven't tried it.

**Comment by jeremybennett** [comment]
> I'll also look at this. I can just try to run both patches on my system as well .
> 
> For gcc tests it uses dejagnu. For that I have a site.exp here. https://github.com/stffrdhrn/or1k-utils
> 
> It's setup to just use sim or user mode qemu. To use or1ksim I think we used telnet before. I haven't tried it.

IIRC or1ksim can be run as a GDB remote target, so you can set up GCC testing via GDB.

**Comment by stffrdhrn** [comment]
I tested on x86 and it works fine.

**Pull request merged**